Source/player.cpp

```diff
@@ -995,22 +984,13 @@ bool PlrHitMonst(int pnum, int m, bool adjacentDamage = false)
 	}
 #endif
 	if ((monster._mhitpoints >> 6) <= 0) {
-		if (monster._mmode == MonsterMode::Petrified) {
-			M_StartKill(m, pnum);
-			monster.Petrify();
-		} else {
-			M_StartKill(m, pnum);
-		}
+		M_StartKill(m, pnum);
 	} else {
-		if (monster._mmode == MonsterMode::Petrified) {
-			M_StartHit(m, pnum, dam);
+		if (!(monster._mmode == MonsterMode::Petrified) && HasAnyOf(player._pIFlags, ItemSpecialEffect::Knockback))
+			M_GetKnockback(m);
+		M_StartHit(m, pnum, dam);
+		if (monster._mmode == MonsterMode::Petrified)
 			monster.Petrify();

```

Comment:
It appears that this was never needed as `M_StartHit()` doesn't appear to change `_mmode`.

Reply:
Actually `M_StartHit()` changes `_mode` to `MonsterMode::HitRecovery` when calling `StartMonsterGotHit`, but it isn't called when monster is alredy Petrified. So you are right :)